### ISSUE TITLE
Assigning the result of a binary expression constructed using a `tsplus` operator to a mutable variable raises a TypeScript error

### ISSUE THREAD

**Issue by IMax153** [ts-plus/typescript#29]
### Description

When assigning the result of a binary expression constructed using a `tsplus` operator expression to a mutable variable (i.e. one created using `let`), a TypeScript error is raised.

For example, the `+` operator is being used to delegate to `Chunk` concatenation via `tsplus`:

```typescript
/**
 * Concatenates two chunks.
 *
 * @tsplus operator ets/Chunk +
 * @tsplus fluent ets/Chunk concat
 */
export function concat_<A, A1>(self: Chunk<A>, that: Chunk<A1>): Chunk<A | A1> {
  return concreteId(self)._concat(concreteId(that))
}
```

However, when I attempt to assign the result of `Chunk` concatenation to a mutable variable using the created `+` operator, a TypeScript error is raised.

![Mutable Assignment Error](https://user-images.githubusercontent.com/38051499/151990285-fed5e052-6e1c-42b5-b518-032df5b9a9d7.png)

In contrast, assigning the result of the binary expression to a `const` variable works as expected.

### Expected Behavior

The result of a binary expression created using a `tsplus` operator should be able to be assigned to a mutable variable.
